storage/sequence_cache.go

```diff
@@ -43,73 +44,96 @@ var errEmptyID = errors.New("empty CommandID used in sequence cache")
 // A SequenceCache is not thread safe. Access to it is serialized
 // through Raft.
 type SequenceCache struct {
-	rangeID roachpb.RangeID
+	rangeID      roachpb.RangeID
+	min, max     roachpb.Key
+	scratchEntry roachpb.SequenceCacheEntry
+	scratchBuf   [256]byte
 }
 
 // NewSequenceCache returns a new sequence cache. Every range replica
 // maintains a sequence cache, not just the leader.
 func NewSequenceCache(rangeID roachpb.RangeID) *SequenceCache {
+	// The sequence number is encoded in decreasing order. We could get away
+	// with zero due to the fact that UUIDs always contain a '4', but let's
```

Comment:
I'm just going to remove this comment. It's about the feasibility of replacing `math.MaxUint32` with zero since no `uuid` is identically `\x00` or `\xff`.